imapclient/imapclient.py

```diff
@@ -565,6 +586,42 @@ def _proc_folder_list(self, folder_data):
             ret.append((flags, delim, name))
         return ret
 
+    def find_special_folder(self, folder_flag):
+        """Try to locate a special folder, like the Sent or Trash folder.
+
+        >>> server.find_special_folder(imapclient.SENT)
+        'INBOX.Sent'
+
+        This function tries its best to find the correct folder (if any) but
+        uses heuristics when the server is unable to precisely tell where are
+        located special folders.
+
+        Returns the name of the folder if found.
+        """
+        # Detect folder by looking for known attributes
+        # Todo: avoid listing all folders by using extended LIST (RFC6154)
+        if self.has_capability('SPECIAL-USE'):
+            for folder in self.list_folders():
+                if folder and len(folder[0]) > 1 and folder[0][1] == folder_flag:
+                    return folder[2]
```

Comment:
`list_folders` should return a namedtuple or something to avoid all these magic indexes. For another PR though...